lib/daemon.js

```diff
@@ -84,12 +84,34 @@ function shutdown() {
     watcher.close();
     watcher = null;
   }
-  server.close(() =>
-    removeConfig(resolver).then(() => {
-      // eslint-disable-next-line n/no-process-exit
-      process.exit(0);
-    })
-  );
+
+  waitForConnections()
```

Comment:
I'm pretty sure you're trying to work around some other issue here. `server.close()` actually waits for any pending connections and doesn't invoke the callback until all connections are gone.

See https://nodejs.org/api/net.html#serverclosecallback.

Reply:
Yes, you're right, finally I've found root cause:
the issue was with `waitForConfig`, for the moment when it was called in `stopDaemon` - it was already removed, so watcher never get notification about

Reply:
Here's correct fix:
https://github.com/mantoni/eslint_d.js/pull/316

Reply:
After #316 merged gonna rebase and cleanup things if you don't mind, otherwise can start new PR

Reply:
Merged. Thank you!